pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -97,16 +98,21 @@ func NewReconciler(mgr manager.Manager, immediateReconcileRequests chan event.Ge
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler.
-func add(mgr manager.Manager, r *DirectReconciler) error {
+func add(mgr manager.Manager, r *DirectReconciler, rolloutPredicate predicate.Predicate) error {
```

Comment:
Nit: `rolloutPredicate` is good in that it says what it is used for, but "rollout" could refer to the resource rolling out or something like that.  So it _might_ be better to call this `additionalPredicate` or `additionalPredicates` or something like that... I'm not sure though!

Reply:
Hmm, I had some trouble deciding on a name for this one. Just changed it to `ReconcileGate` / `ReconcilePredicate`, wdyt?